module/validation/seal_validator.go

```diff
@@ -183,6 +184,17 @@ func (s *sealValidator) Validate(candidate *flow.Block) (*flow.Seal, error) {
 		ancestorID = ancestor.ParentID
 	}
 
+	// since the payload has passed the receiptExtend, the receipts must be incorporated.
+	// we must adding them to the incorporatedResults, otherwise if a matching receipt
+	// and seal are added in the same block, the seal would be considered as invalid by mistake.
+	for _, receipt := range candidate.Payload.Receipts {
+		resultID := receipt.ExecutionResult.ID()
+		incorporatedResults[resultID] = flow.NewIncorporatedResult(
+			receipt.ExecutionResult.BlockID,
+			&receipt.ExecutionResult,
+		)
+	}
+
```

Comment:
Would prefer no to add logic, which we will have to remove right away for the next sealing phase.